### PR TITLE
Week 2 - Class Rational

### PR DESCRIPTION
**Pull request by vadimkatsel**

### CODE REVIEW

File: scala-course-2022-autumn/src/main/scala/karazin/scala/users/group/week2/homework/Homework.scala [hunk 1]
```diff
@@ -34,28 +35,36 @@ object Homework:
...
+    override def toString: String =
+      s"${this.numer}/${this.denom}"
 
     private def gcd(a: Int, b: Int): Int =
       if b == 0 then a else gcd(b, a % b)
 
     private lazy val g = gcd(abs(x), y)
 
-    override def equals(other: Any): Boolean = ???
+    override def equals(other: Any): Boolean =
```
Comment (IgorWolkov) [inline scala-course-2022-autumn/src/main/scala/karazin/scala/users/group/week2/homework/Homework.scala:66]: Please review https://www.baeldung.com/java-equals-hashcode-contracts

**Review by IgorWolkov — CHANGES_REQUESTED** [review]
Almost done, please fix the issue with hashcode.

**Review by IgorWolkov — APPROVED** [review]
Well done!

**Comment by vadimkatsel** [comment]
Can this HW be merged?

**Pull request merged**